---
title: "Installing the RSAT Docker container"
author: "Najla Ksouri, Bruno Contreras Moreira"
date: '`r Sys.Date()`'
output:
  html_document:
    self_contained: no
    fig_caption: yes
    highlight: zenburn
    number_sections: yes
    theme: cerulean
    toc: yes
    toc_depth: 4
    toc_float: yes
  pdf_document:
    fig_caption: yes
    highlight: zenburn
    number_sections: yes
    toc: yes
    toc_depth: 4
  word_document: default
editor_options: 
  chunk_output_type: console
#css: ../../course.css
---


```{r setup, include=FALSE}
library(knitr, warn.conflicts = FALSE)

knitr::opts_chunk$set(
  echo = TRUE, 
  eval = FALSE, 
  cache = TRUE, 
  message = FALSE, 
  warning = FALSE, 
  comment = "",  
  fig.align = "center",
  fig.width = 7, 
  fig.height = 5,
  out.width = "90%",
  fig.path = "figures/")
```

****************************************************************
# Introduction

This page explains how to install a Docker container of the Regulatory Sequence Analysis Tools. 

# Requirements

## Availability

RSAT Docker containers are available at https://hub.docker.com/r/biocontainers/rsat/tags 

Each release corresponds to a tag from https://github.com/rsa-tools/rsat-code/tags

## Operating systems

Docker containers have so far been tested on Linux, Windows and macOS (Intel) systems. 

Please let us know or send us a PR at https://github.com/rsa-tools/installing-RSAT if you succeed in running it on other settings.

## Docker 

To run containers you must have Docker installed in your system. 

You can find instructions for installing the Docker (Server) engine on **Linux** at https://docs.docker.com/engine/install/#server .

On **Windows**, our recommended procedure is to i) install the Windows Subsystem for Linux (WSL) and then ii) the Docker engine:

 * The installation of WSL is simple, as it you can fetch it from the Microsoft Store for free (read more [here](https://learn.microsoft.com/en-us/windows/wsl/install) and 
[here](https://devblogs.microsoft.com/commandline/the-windows-subsystem-for-linux-in-the-microsoft-store-is-now-generally-available-on-windows-10-and-11/)).
 * The installation of Docker engine within WSL is explained at https://dev.to/felipecrs/simply-run-docker-on-wsl2-3o8 and https://docs.docker.com/engine/install/ubuntu .

Our community have also succeeded in running the RSAT container with [Docker Desktop](https://www.docker.com/products/docker-desktop)
for **MacOS** (also available for Linux and Windows systems).

<!--
$> docker -v
Docker version 24.0.2, build cb74dfc

$> system_profiler SPSoftwareDataType SPHardwareDataType
Software:
    System Software Overview:
      System Version: macOS 11.7.2 (20G1020)
      Kernel Version: Darwin 20.6.0
      Boot Volume: Macintosh HD
      Secure Virtual Memory: Enabled
-->

<!--
**Note:** see [instructions](https://stackoverflow.com/questions/61530874/docker-how-do-i-add-myself-to-the-docker-users-group-on-windows) to add your Windows user to the 'docker-users' group, you will need admin privileges. -->

****************************************************************

# Installation instructions

If you want to fetch a ready-to-run Docker container with RSAT please select a tagged version from https://hub.docker.com/r/biocontainers/rsat/tags . 

The steps below are for version **rsat:20240507_cv1**, you might change it as needed for other versions:

 1. Download a container (this will take 10GB of your filesystem):

        # from the Linux/WSL terminal
        docker pull biocontainers/rsat:20240507_cv1
        
    <!--This can also be done from the Docker Desktop app, as shown in the figure:
    ![](./DockerDesktopWindows.png) -->
        

 2. Create local folders for input data and results, outside the container. In the example two folders are created, named *rsat_data/* and *rsat_results/*, which are placed in the current location ($HOME). In addition, a subfolder (*rsat_data/genomes*) is also created and permissions are changed so that anyone can write there. Note those command lines will work equally on Linux/Windows systems. **Note:** you can place these folders anywhere in your system, but please check their paths and modify them in **step 3** accordingly:
 
        # assuming you are at $HOME 
        mkdir -p rsat_data/genomes rsat_results
        chmod -R a+w rsat_data rsat_results

 3. Launch Docker RSAT container. Note that the local folders from **step 2** are mounted as volumes in the container. If you changed their locations please adjust their paths to the right of the colons. **Note:** after this instruction, all other commands should be typed and executed at the container's terminal. :

        docker run --rm -v $HOME/rsat_data:/packages/rsat/public_html/data/ -v $HOME/rsat_results:/home/rsat_user/rsat_results -it biocontainers/rsat:20240507_cv1
        
        # you should see a warning that can be safely ignored: 
        #
        # * Starting Apache httpd web server apache2
        # (13)Permission denied: AH00091: apache2: could not open error log file /var/log/apache2/error.log.
        # AH00015: Unable to open logs
        # Action 'start' failed.
        # The Apache error log may have more information


 4. Download organism from public RSAT server, such as the Plants server. Other available servers are http://fungi.rsat.eu, http://metazoa.rsat.eu, http://protists.rsat.eu and http://teaching.rsat.eu

        download-organism -v 2 -org Prunus_persica.Prunus_persica_NCBIv2.38 -server https://rsat.eead.csic.es/plants

 5. Testing:

        cd rsat_results 
        make -f ../test_data/peak-motifs.mk RNDSAMPLES=2 all

 6. To install any organism, please follow the instructions at [managing-RSAT](https://rsa-tools.github.io/managing-RSAT/genome_installation/install_organisms_FASTA_GTF.html).

 7. To connect to RSAT Web server running from Docker container (Linux only): 

        # to start the Docker Web server launch the container and do (check password in Dockerfile):
        sudo service apache2 restart
        hostname -I # should return IP address

        # finally open the following URL in your browser, using the IP address, ie http://172.17.0.2/rsat
 
# Complete interactive tutorial

Once you installed the RSAT Docker container you can follow our protocol on using a Docker container interactively to carry out motif analysis in co-expression networks at:

https://eead-csic-compbio.github.io/coexpression_motif_discovery/peach/Tutorial.html 

# Calling RSAT tools non-interactively

In addition to logging into the Docker container as explained in the previous section,
you can also call individual tools from the terminal non-interactively. See the example below with
[*peak-motifs*](https://doi.org/10.1093/nar/gkr1104) and a user-provided FASTA file (`test.fa`) in the current directory:

    docker run -v $PWD:/home/rsat_user -v $PWD/out:/home/rsat_user/out biocontainers/rsat:20240507_cv1 peak-motifs -help

    docker run -v $PWD:/home/rsat_user -v $PWD/out:/home/rsat_user/out biocontainers/rsat:20240507_cv1 peak-motifs -i test.fa -outdir out -prefix test

A more ellaborated example follows, were discovered motifs are compared to the collection of plant DNA motifs from footprintDB; list all available motif collections at https://github.com/rsa-tools/motif_databases :

    docker run -v $PWD:/home/rsat_user -v $PWD/out:/home/rsat_user/out biocontainers/rsat:20240507_cv1 peak-motifs -i test.fa -outdir out -prefix test -motif_db footDB transfac  /packages/rsat/public_html/motif_databases/footprintDB/footprintDB.plants.motif.tf

Note these examples require binding two folders, but this might change with other tools:

* The current folder from where the container is called ($PWD). This allows processing any FASTA file in the current directory.
* The folder to save the results (out/). This should point to an existing local folder where anybody can write (chmod 777).

# Dockerfiles

Dockerfiles for the containers at [biocontainers/rsat](https://hub.docker.com/r/biocontainers/rsat) 
can be found at https://github.com/BioContainers/containers/tree/master/rsat

Those are based on a standalone Dockerfile, available at https://github.com/rsa-tools/rsat-code/tree/master/docker, which we use for testing.

